waku/message/v1/message.proto

```diff
@@ -0,0 +1,12 @@
+syntax = "proto3";
+
+// 14/WAKU2-MESSAGE rfc: https://rfc.vac.dev/spec/14/
+package waku.message.v1;
+
+message WakuMessage {
+  bytes payload = 1;
+  string content_topic = 2;
+  uint32 version = 3;
+  optional sint64 timestamp = 10;
+  optional bool ephemeral = 31;
```

Comment:
I was going to say that `optional` is mostly useless for `proto3` for fields of types that can be `null`.

However, it looks like the definition changed recently: https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules

> - `singular`: a well-formed message can have zero or one of this field (but not more than one). When using proto3 syntax, this is the default field rule when no other field rules are specified for a given field. **You cannot determine whether it was parsed from the wire**. **It will be serialized to the wire unless it is the default value.** For more on this subject, see [Field Presence](https://github.com/protocolbuffers/protobuf/blob/main/docs/field_presence.md).
> - `optional`: the same as singular, except that **you can check to see if the value was explicitly set**. An optional field is in one of two possible states:
>   - the field is set, and **contains a value that was explicitly set or parsed from the wir**e. It will be serialized to the wire.
>   - the field is unset, and will **return the default value**. It will **not be serialized to the wire.**

This is news to me and I am pretty sure they changed that recently.

Just wanted to be sure we are on the same page.

Cc @richard-ramos @danisharora099 FYI

Reply:
This is definitely different from what I understood before. Previously the understanding was that all fields are `optional`, as you say. My understanding is then that field presence checks are the only difference - @LNSD why would `ephemeral` then be `optional`? If not present, we'll just assume it to be `false`, which is expected.

Reply:
FYI, there is also this other protobuf language specification from the guys at [buf.build]() that @richard-ramos shared some time ago: https://protobuf.com/docs/language-spec

THe relevant section is the ["Field Presence and Default Values"](https://protobuf.com/docs/language-spec#field-presence-and-default-values).

-----

I knew that this was something that would catch your attention 😁 

> Why be **explicit** with the `optional` annotation?

Some languages have the "optional" type (e.g., Rust, Nim, Java), and this modifier instructs the code generation tools to emit an `Option[bool]` type for that specific field.

Reply:
Indeed, and it's a great addition to the proto syntax imo. It makes perfect sense to be explicitly `optional` for `timestamp` - I'm just not sure that `ephemeral` is truly optional in the same way? In other words, there is no semantic difference between `ephemeral.isNone()` and `ephemeral.get() == false`. It adds an unnecessary state.

Reply:
> @fryorcraken: This is news to me and I am pretty sure they changed that recently.

This is very interesting! According to internet archive, it seems this change in the language guide was introduced somewhere between Sep14,2022 to Oct4,2022. It's a nice because previously the behavior for this kind of fields was not defined and left up to implementations to determine what to do.

Reply:
> FYI, there is also this other protobuf language specification from the guys at [buf.build](https://github.com/vacp2p/waku/pull/1) that @richard-ramos shared some time ago: https://protobuf.com/docs/language-spec

I have found this explanation less clear that the original proto3 specs :)
I think the buf.build spec is now deprecated because it attempts to describe the observed behaviour of proto3, before spec update, which was: default = absent = zero value. Which meant: It is not possible to detect presence of a field that has a zero value.

Whereas now, with the `optional` keyword, we are saying that it **is** possible to detect presence of a field that has a zero value.
This changes things quite a bit and has @jm-clius pointed out, I would agree that:

- `ephemeral` should not be set as `optional`: a value of `null` or `false` has a same functional consequence so there is no need to differentiate them.
- `timestamp`: this is interesting now because we define our software around _no timestamp means timestamp = 0_ but we could change that. If we maintain the current behaviour then the `optional` keyword should be removed from `timestamp` too.